### PR TITLE
Add eus 4.10 to 4.12 upgrade test on gcp azure and vsphere

### PR DESCRIPTION
**Pull request by shellyyang1989**
Prior to the change, only aws-ipi-ovn is covered on the 4.10 to 4.12 eus upgrade. The change extends the test as below:

amd64-nightly-4.12-upgrade-from-eus-4.10 on vsphere-ipi-p2-f14 and azure-ipi-p2-f14
amd64-stable-4.12-upgrade-from-eus-4.10 on gcp-ipi-ovn-ipsec-p2-f14
arm64-nightly-4.12-upgrade-from-eus-4.10 on aws-ipi-ovn-p2-f14

### CODE REVIEW

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-eus-4.8-gcp-ipi-p2-f28

**Comment by shellyyang1989** [comment]
@liangxia @jianlinliu @evakhoni please help review. Thanks!

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__arm64-stable-4.12-upgrade-from-eus-4.10.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      architecture: arm64
+      channel: candidate
+      version: "4.12"
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: azure-ipi-p2-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__arm64-stable-4.12-upgrade-from-eus-4.10.yaml:76]: I think azure arm get supported from 4.12, but not 4.10.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-eus-4.10.yaml [hunk 2]
```diff
@@ -40,43 +40,42 @@ releases:
   intermediate:
     release:
       architecture: amd64
-      channel: fast
+      channel: candidate
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-eus-4.10.yaml:43]: Here the `candidate` channel means OCP official channel, right? If I remember right, the order is like `candidate` -> `fast` -> `stable`, right?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__arm64-stable-4.12-upgrade-from-eus-4.10.yaml (hunk 1 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__arm64-stable-4.12-upgrade-from-eus-4.10.yaml:76]: ah, thanks for pointing it out. Seems like only aws is supported in 4.10 arm64. Let me update it.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-eus-4.10.yaml (hunk 2 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-eus-4.10.yaml:43]: Candidate channel is not something we recommend customers to use. It includes all the engineering candidate, release candidate and standard/GA releases.

> the order is like candidate -> fast -> stable

That's right

File: ci-operator/step-registry/openshift/upgrade/qe/test/paused/openshift-upgrade-qe-test-paused-chain.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+chain:
+  as: openshift-upgrade-qe-test-paused
```
Comment (liangxia) [inline ci-operator/step-registry/openshift/upgrade/qe/test/paused/openshift-upgrade-qe-test-paused-chain.yaml:2]: This chain has exactly the same content as in `cucushift-upgrade-paused`, do we need to duplicate like this ?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml [hunk 4]
```diff
@@ -64,21 +60,37 @@ resources:
...
+      E2E_RUN_INTERMEDIATE_TAGS: '@4.11 and @vsphere-ipi and @upgrade-sanity and not
+        @fips'
+      E2E_RUN_LATEST_TAGS: '@4.12 and @vsphere-ipi and @upgrade-sanity and not @fips'
+      UPGRADE_CHECK_RUN_TAGS: '@4.10 and @vsphere-ipi and @upgrade-check and not @fips'
+      UPGRADE_PRE_RUN_TAGS: '@4.10 and @vsphere-ipi and @upgrade-prepare and not @fips'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test-paused
+    workflow: cucushift-installer-rehearse-vsphere-ipi
+- as: azure-ipi-p2-f14
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:78]: Could you help to sort the tests ?

File: ci-operator/step-registry/openshift/upgrade/qe/test/paused/openshift-upgrade-qe-test-paused-chain.yaml (hunk 3 above)
Comment (shellyyang1989) [reply ci-operator/step-registry/openshift/upgrade/qe/test/paused/openshift-upgrade-qe-test-paused-chain.yaml:2]: Our upgrade chain and arch migration chain are in the openshift/upgrade/qe/test, so moving EUS upgrade to the same folder to make them unified. Does it look good to you?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml (hunk 4 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:63]: ah, thanks for pointing it out. Updated.

File: ci-operator/step-registry/openshift/upgrade/qe/test/paused/openshift-upgrade-qe-test-paused-chain.yaml (hunk 3 above)
Comment (liangxia) [reply ci-operator/step-registry/openshift/upgrade/qe/test/paused/openshift-upgrade-qe-test-paused-chain.yaml:2]: Instead of copy, I'd prefer move.

Comment (shellyyang1989) [reply ci-operator/step-registry/openshift/upgrade/qe/test/paused/openshift-upgrade-qe-test-paused-chain.yaml:2]: Removed cucushift-upgrade-paused chain and updated the tests which was using cucushift-upgrade-paused to openshift-upgrade-qe-test-paused

**Pull request merged**